### PR TITLE
Log the percentage of optimizable keyword argument methods

### PR DESCRIPTION
**Pull request by wildmaples**
### Why 
The goal is to use this build to run SFR to see what the optimizable % looks like. 

See: https://github.com/Shopify/truffleruby-storefront-renderer/issues/117

⚠️ This change is an experiment and not meant to be shipped!!!! (Ignore the lint issue on CI) 

### Testing

Using this test script: 

```ruby
def optimizable(a:); end
def not_optimizable(a:, **rest); end

100.times do
  optimizable(a: 1)
  not_optimizable(a: 1, b:2)
end

10_000.times do
  optimizable(a: 1)
end
```

Generates: 

Screen Shot 2021-06-22 at 4 22 27 PM

----------------------------

Without the exaggerated 10_000 optimized run....

```ruby
def optimizable(a:); end
def not_optimizable(a:, **rest); end

100.times do
  optimizable(a: 1)
  not_optimizable(a: 1, b:2)
end
```

Screen Shot 2021-06-22 at 4 23 14 PM

### CODE REVIEW

**Comment by chrisseaton** [comment]
Probably run on `bench/boot/request`, but you need to comment out the signal line at the end of it, so you can get the normal exit and your logging output.

You could even then modify to run 2 request and see the diff in the numbers and so report pre-request.

**Comment by nirvdrum** [comment]
The phrasing "Total dynamic keyword arg calls" seems inaccurate. As far as I can tell, that count is actually the total number of method calls, with or without keyword arguments. If I'm just reading that wrong, feel free to ignore. As long as you know what the actual value represents, I think you're fine.

**Comment by wildmaples** [comment]
Thank you both for the feedback!

> We should either measure '% of keyword argument calls that we will optimise' or '% of calls that we will optimise'. It's currently the latter but worded like the former.

I'll add both % cause the more information the better! Will push another commit with the change (and correct the wording).

**Comment by wildmaples** [comment]
Alright screenshot below is for the latest commit [`a7366c8` (#8)](https://github.com/Shopify/truffleruby/pull/8/commits/a7366c8bad140a84ff158bee5d974a23794dc5d9)

The first run is without 10_000 times optimizable and the latter with 10_000 times optimizable. You can see that the # of method calls increased correctly. I've also changed the names to be more accurate to what we're actually measuring. 

Screen Shot 2021-06-22 at 9 26 45 PM

**Comment by chrisseaton** [comment]
99% of keyword methods calls and **54%** of all method calls? Half of all non-AST-inlined method calls (so not very basic operators) have keyword arguments? That's absolutely crazy. This could be massive for startup time, compilation time, and maybe really good for performance.

I'll have to have a read and think about how there can be so many keyword argument calls.

File: src/main/java/org/truffleruby/RubyContext.java [hunk 1]
```diff
@@ -22,6 +22,7 @@
 import java.util.WeakHashMap;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicLong;
```
Comment (tomstuart) [inline src/main/java/org/truffleruby/RubyContext.java:25]: I know you told us to ignore the linting failures, but: why is this import here?

File: src/main/java/org/truffleruby/RubyContext.java [hunk 2]
```diff
@@ -52,6 +53,7 @@
 import org.truffleruby.core.inlined.CoreMethods;
 import org.truffleruby.core.kernel.AtExitManager;
 import org.truffleruby.core.kernel.TraceManager;
+import org.truffleruby.core.method.RubyMethod;
```
Comment (tomstuart) [inline src/main/java/org/truffleruby/RubyContext.java:56]: Ditto.

**Pull request closed without merging**